### PR TITLE
Location routing query

### PR DESCRIPTION
**Pull request by ashish-sharma09**

### CODE REVIEW

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy [hunk 1]
```diff
@@ -35,27 +38,29 @@ class PipeReadAuthenticationProviderIntegrationSpec extends Specification {
...
+                      users:
+                        $USERNAME:
+                          password: $PASSWORD
+                          roles:
+                            - PIPE_READ
+                        $RUNSCOPE_USERNAME:
+                          password: $RUNSCOPE_PASSWORD
+                          roles:
+                            - PIPE_READ
+                    """
```
Comment (ashish-sharma09) [inline pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy:59]: just formatting here

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy [hunk 2]
```diff
@@ -35,27 +38,29 @@ class PipeReadAuthenticationProviderIntegrationSpec extends Specification {
     @Shared LocationResolver locationResolver = Mock()
 
     void setupSpec() {
+        locationResolver.resolve(_) >> [new Cluster("cluster1")]
```
Comment (ashish-sharma09) [inline pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy:41]: location resolver expectation added

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerBatchIntegrationSpec.groovy [hunk 3]
```diff
@@ -1,86 +1,63 @@
...
 
-import java.nio.charset.StandardCharsets
+import javax.inject.Inject
+import javax.inject.Named
 
+import static java.util.OptionalLong.of
 import static org.hamcrest.Matchers.equalTo
 
 @Newify(Message)
+@MicronautTest
```
Comment (ashish-sharma09) [inline pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerBatchIntegrationSpec.groovy:18]: Modified this test to use MicronautTest framework

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java [hunk 4]
```diff
@@ -184,33 +191,39 @@ private static String getSelectEventsWithTypeFilteringQuery(final long maxBatchS
...
+                  withInnerJoinToClusters() +
+            "   AND events.msg_offset >= ? " +
+            "   AND type = ANY (string_to_array(?, ','))" +
+            " ORDER BY msg_offset " +
+            " LIMIT ?" +
             " ) unused " +
             " WHERE running_size <= " + maxBatchSize;
     }
 
+    private String withInnerJoinToClusters() {
```
Comment (LorenzoCianciaruso) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java:220]: I do see why this has been extracted although Im not sure if it make in any better in term of readability

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.tesco.aqueduct.pipe.http
```
Comment (LorenzoCianciaruso) [inline pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy:1]: Is this class needed?

File: pipe-http-client/src/test/resources/logback-test.xml [hunk 6]
```diff
@@ -6,8 +6,6 @@
         </encoder>
     </appender>
 
-    <logger name="io.micronaut.http.client" level="DEBUG"/>
```
Comment (adam-yorwerth) [inline pipe-http-client/src/test/resources/logback-test.xml:9]: Any reason that we removed this?

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy (hunk 5 above)
Comment (adam-yorwerth) [reply pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy:1]: It's vital to the codebase Lorenzo ! 😂

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy (hunk 1 above)
Comment (sebastianlavender) [reply pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadAuthenticationProviderIntegrationSpec.groovy:59]: Formatting looks good to me

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java (hunk 4 above)
Comment (sebastianlavender) [reply pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java:220]: I don't think it makes the readability worse but it does ensure we are following DRY better

File: pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy (hunk 5 above)
Comment (ashish-sharma09) [reply pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy]: :-) We initially created the test with this name using MicronautTest framework and renamed later to PipeReadControllerIntegartionSpec but compiler kept complaining with PipeReadController ClassNotFound ¯\_(ツ)_/¯ so we had to keep this in place to keep the compiler happy. Going to remove this now and push.

Comment (sebastianlavender) [reply pipe-http-server/src/integration/groovy/com/tesco/aqueduct/pipe/http/PipeReadControllerSpec.groovy]: It was needed for another test to run due to it complaining about the class not existing - it has now been removed as the test functions without it

File: pipe-http-client/src/test/resources/logback-test.xml (hunk 6 above)
Comment (ashish-sharma09) [reply pipe-http-client/src/test/resources/logback-test.xml:9]: I think this was put in place during identity issue token 404 error debugging. It just makes test logs verbose hence removed it, can be reinstated as necessary.

**Pull request merged**